libs/blocks/email-collection/email-collection.js

```diff
@@ -206,13 +208,16 @@ async function submitForm(form) {
       timezoneOffset: -date.getTimezoneOffset(),
     };
 
-    const { error, data } = await runtimePost(
+    const { error, data, status } = await runtimePost(
       getApiEndpoint(),
       bodyData,
       ['occupation', 'organization', 'state'],
     );
 
-    if (error) messageParams.errorMsg = error;
+    if (error) {
+      await redirectToSignIn(dialog, status === 401);
```

Comment:
I would imagine redirectToSignIn is only called if the status is a 401... shouldn't you move the check up?

```suggestion
    if (error) {
      if(status === 401) await redirectToSignIn(dialog);
```